File: ios-swiftui/library/Sources/PlanetComponents/TspInput.swift
import SwiftUI

public struct TspInput: View {
    @Binding var value: String
    let placeholder: String
    let variant: String
    let disabled: Bool
    let theme: StarPlanetTheme

    public init(value: Binding<String>, placeholder: String = "", variant: String = "default", disabled: Bool = false, theme: StarPlanetTheme = .sky) {
        self._value = value
        self.placeholder = placeholder
        self.variant = variant
        self.disabled = disabled
        self.theme = theme
    }

    public var body: some View {
        TextField(placeholder, text: $value)
            .disabled(disabled)
            .padding(.horizontal, 14)
            .frame(minHeight: 48)
            .foregroundColor(theme.textPrimary)
            .background(theme.surfaceRaised)
            .overlay(RoundedRectangle(cornerRadius: 16).stroke(variant == "error" ? theme.danger : theme.borderDefault, lineWidth: 1))
            .clipShape(RoundedRectangle(cornerRadius: 16))
            .opacity(disabled ? 0.45 : 1)
    }
}

File: flutter/library/lib/src/tsp_input.dart
import 'package:flutter/material.dart';
import 'star_planet_theme.dart';

class TspInput extends StatelessWidget {
  const TspInput({super.key, required this.controller, this.placeholder = '', this.variant = 'default', this.disabled = false, this.theme = StarPlanetTheme.sky});
  final TextEditingController controller;
  final String placeholder;
  final String variant;
  final bool disabled;
  final StarPlanetTheme theme;

  @override
  Widget build(BuildContext context) => TextField(
        controller: controller,
        enabled: !disabled,
        style: TextStyle(color: theme.textPrimary),
        decoration: InputDecoration(
          hintText: placeholder,
          filled: true,
          fillColor: theme.surfaceRaised,
          contentPadding: const EdgeInsets.symmetric(horizontal: 14, vertical: 14),
          constraints: const BoxConstraints(minHeight: 48),
          border: OutlineInputBorder(borderRadius: BorderRadius.circular(16), borderSide: BorderSide(color: variant == 'error' ? theme.danger : theme.borderDefault)),
          enabledBorder: OutlineInputBorder(borderRadius: BorderRadius.circular(16), borderSide: BorderSide(color: variant == 'error' ? theme.danger : theme.borderDefault)),
          disabledBorder: OutlineInputBorder(borderRadius: BorderRadius.circular(16), borderSide: BorderSide(color: theme.borderDefault)),
          focusedBorder: OutlineInputBorder(borderRadius: BorderRadius.circular(16), borderSide: BorderSide(color: theme.brandPrimary)),
        ),
      );
}
